File: leetcode/src/java/BinaryTreePostorderTraversal.java
import java.util.ArrayList;
import java.util.List;

public class BinaryTreePostorderTraversal extends BaseTreeNode {
    public static void main(String[] args) {
        BinaryTreePostorderTraversal instance = new BinaryTreePostorderTraversal();
        System.out.println(instance.postorderTraversal(generateTreeNode(new Integer[]{1, null, 2, 3})));
    }
    public List<Integer> postorderTraversal(TreeNode root) {
        List<Integer> result = new ArrayList<Integer>();
        postorderTraversal(root, result);
        return result;
    }
    private void postorderTraversal(TreeNode node, List<Integer> previousResult) {
        if (node != null) {
            postorderTraversal(node.left, previousResult);
            postorderTraversal(node.right, previousResult);
            previousResult.add(node.val);
        }
    }
}

File: leetcode/src/python/binary_tree_postorder_traversal.py
from typing import Optional
from typing import List

# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right
class BinaryTreePostorderTraversal:
    def postorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        result = []
        if root is not None:
            result += self.postorderTraversal(root.left)
            result += self.postorderTraversal(root.right)
            result.append(root.val)
        return result